actions/ci-beholder-validator/action.yml

```
@@ -0,0 +1,105 @@
+name: "Schema Validator Action"
+description: "Validates schema compatibility with schema registry"
+
+inputs:
+  # general inputs
+  checkout-repo-fetch-depth:
+    description: "number of commits to fetch"
+    required: false
+    default: "0"
+  # image inputs
+  docker-registry:
+    description: "Docker registry to use"
+    required: true
+    default: "aws"
+  image-tag:
+    description: "Image tag to use"
+    required: false
+    default: "latest"
+  # aws inputs
+  aws-role-duration-seconds:
+    description: ""
+    required: false
+    default: "900"
+  aws-region:
+    description: ""
+    required: false
+  aws-account-number:
+    description: ""
+    required: false
+  aws-role-arn:
+    description: ""
+    required: false
+
+runs:
+  using: composite
+  steps:
+    - name: Checkout repo
+      uses: actions/checkout@v4.2.1
+      with:
+        fetch-depth: 0
+    - name: Detect changed files
+      id: changes
+      shell: bash
+      run: |
+        git fetch --prune origin +refs/heads/*:refs/remotes/origin/*
+        # TODO: This logic need to be moved to code
+        changed_files=$(git diff --name-only origin/${{github.event.repository.default_branch }})
```

Could this be set as an env variable eg 
```
shell: bash
env:
  DEFAULT_BRANCH: ${{ github.event.repository.default_branch }}
```

and then used like this
```suggestion
        changed_files=$(git diff --name-only origin/${{ env.DEFAULT_BRANCH }})
```
Issue [attacker-controllable code ](https://openssf.org/blog/2024/08/12/mitigating-attack-vectors-in-github-workflows/#:~:text=Code%20injection%20on%20GitHub%20Actions,permissions%20granted%20to%20the%20workflow.).

---

Replied above.

---

The entire code will be running in runner and would be interacting with a redpanda schema registry brought through docker. Do you see any security vulnerabilities with this in that case?

---

Any `${{ ... }}` used in Github Actions shell is dangerous because it's unchecked string replacement. This could technically cause arbitrary code execution.

For example, if the default branch is `main);echo "hello world"`  (brutally obvious example).

Then this  would effectively be:
```
changed_files=$(git diff --name-only origin/main);echo "hello world"
```

---

This is resolved